sys/dev/sound/pci/ich.c

```diff
@@ -1196,7 +1196,7 @@ ich_pci_resume(device_t dev)
 
 	ICH_LOCK(sc);
 	/* Reinit audio device */
-    	if (ich_init(sc) == -1) {
+  if (ich_init(sc) != 0) {
```

Comment:
Perhaps ideally the error from `ich_init` would be returned directly, but this fix is also fine.

Reply:
yes, I agree with you. I only did one line change since it is returning ENXIO anyway.